src/openfermion/utils/_hubbard_trotter_error.py

```diff
@@ -0,0 +1,142 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""Module to compute Trotter errors in the plane-wave dual basis."""
+from __future__ import absolute_import
+from future.utils import iteritems, itervalues
+
+import numpy
+
+from openfermion.config import *
+from openfermion.hamiltonians import fermi_hubbard
+from openfermion.ops import FermionOperator, normal_ordered
+from openfermion.utils import count_qubits
+from openfermion.utils._dual_basis_trotter_error import (
+    dual_basis_error_bound, dual_basis_error_operator, stagger_with_info)
+
+
+def fermi_hubbard_error_operator(terms, indices=None, is_hopping_operator=None,
+                                 verbose=False):
+    """Determine the difference between the exact generator of unitary
+    evolution and the approximate generator given by the second-order
+    Trotter-Suzuki expansion for the Hubbard model.
+
+    Args:
+        terms: a list of FermionOperators in the Hamiltonian in the
+               order in which they will be simulated.
+        indices: a set of indices the terms act on in the same order as terms.
+        is_hopping_operator: a list of whether each term is a hopping operator.
+        verbose: Whether to print percentage progress.
+
+    Returns:
+        The difference between the true and effective generators of time
+            evolution for a single Trotter step.
+
+    Notes: follows Equation 9 of Poulin et al.'s work in "The Trotter Step
+        Size Required for Accurate Quantum Simulation of Quantum Chemistry".
+    """
+    return dual_basis_error_operator(
+        terms, indices=indices, is_hopping_operator=is_hopping_operator,
+        verbose=verbose)
+
+
+def fermi_hubbard_error_bound(terms, indices=None, is_hopping_operator=None,
```

Comment:
This function is specific to the second order Trotter formula. Somehow that should be reflected in the name of the function. Also, what is it that makes this code specific to the Fermi-Hubbard model? Is it simply that it anticipates only having a^\dagger_p a_q and n_p n_q terms? Because that form also appears for the dual basis.

Reply:
This code always uses the same restrictions on the dual_basis_error_operator code (forcing several optional variables to their defaults) so I think it's nice to have for that simplicity.